### DISCUSSION TITLE
Transparency Issue with BMP Textures in <model-viewer> on iOS

### DISCUSSION THREAD

**Discussion by valentinbodur** [google/model-viewer#4884] [Q&A]
Hello,

I'm experiencing an issue with displaying PNG textures with transparency in the <model-viewer> on iOS devices. The texture appears correctly with full transparency on Android, but on iOS, it adds a white tint to the transparent areas.

Here are the specifics of my setup:
Modeling Software: Blender 4.2
Model Viever 3.5.0
Texture Setup: I have a PNG texture connected to the Principled BSDF shader. The color is linked to the color input, and alpha is connected to the alpha input.
Issue Description: The transparency, which works perfectly on Android, doesn't display correctly on iOS. Instead of being transparent, the areas that are supposed to be clear show a white tint.
Attached Image: I have attached an image that shows how the transparency issue appears in AR on iOS. The model made in Blender consists of two planes (same metallic and roughness) placed closely together; the front one features a texture with transparency.

![test transparency ar ios](https://github.com/user-attachments/assets/34883276-5be8-4a7c-9795-54685b4c2daf)

Any advice on how to handle this transparency issue on iOS would be greatly appreciated!

Thank you!

**Comment by elalish** [comment]
Hmm, glTF doesn't support BMP, so I'd guess Blender converted it to PNG or something - I doubt that's the problem. Two nearly coplanar surfaces is a much more likely problem - z-fighting etcetera. The biggest problem is probably just USDZ conversion, since Apple *still* doesn't support glTF - these formats are not entirely compatible and the transform is lossy. You may want to experiment a bit to see what works best.

**Reply by valentinbodur** [reply]
You are right, it was about PNG files not BMP, I corrected it in the post.
I will try to increase the distance between the planes a little to see if the strange effect on transparency still appears and I will update with the result.

**Reply by valentinbodur** [reply]
I moved the plans and the same effect. on iPhone 11 the transparency is not complete, see in the first image.
in the second image the same model on the same model-viewer page but on android the transparency is total.
![ios](https://github.com/user-attachments/assets/6c593892-5228-416e-9c11-9debb3350e87)
![android](https://github.com/user-attachments/assets/77ef328e-9caa-496c-a6af-f59d416b64c3)

**Reply by elalish** [reply]
I would guess this is because glTF has two types of transparency: alpha (coverage) and transmission (glass). I think USDZ only has alpha, but they may be hacking it to look slightly more glass like. Hence why the underspecified USDZ is such a problem format.
